src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -383,6 +389,43 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
             }
         }
 
+        /// <summary>
+        /// Returns true if tuple type syntax can be used to refer to the tuple type without loss of information.
+        /// For example, it cannot be used when extension tuple is using non-default friendly names. 
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <returns></returns>
+        private bool CanUseTupleTypeName(INamedTypeSymbol symbol)
+        {
+            INamedTypeSymbol currentUnderlying = symbol.TupleUnderlyingType;
+
+            if (currentUnderlying.Arity == TupleTypeSymbol.RestPosition)
+            {
+                do
+                {
+                    var currentTuple = (INamedTypeSymbol)currentUnderlying.TypeArguments[TupleTypeSymbol.RestPosition - 1];
+                    Debug.Assert(currentTuple.IsTupleType);
+
+                    var elementNames = currentTuple.TupleElementNames;
+                    if (!elementNames.IsDefault)
+                    {
+                        for (int i = 0; i < elementNames.Length; i++)
+                        {
+                            if (elementNames[i] != TupleTypeSymbol.TupleMemberName(i + 1))
+                            {
+                                return false;
+                            }
+                        }
+                    }
+
+                    currentUnderlying = currentTuple.TupleUnderlyingType;
+                }
+                while (currentUnderlying.Arity == TupleTypeSymbol.RestPosition);
+            }
+
+            return true;
+        }
+
```

Comment:
Not blocker for this PR, but I wonder if there are other cases when we should fall back to displaying the underlying type. See issue #10808 for instance.